src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1857,6 +1858,9 @@ public void makeIndexNonQueryable(Index index, Index.Status status)
 
     public void makeIndexQueryable(Index index, Index.Status status)
     {
+        if (CassandraRelevantProperties.SAI_INDEX_READS_DISABLED.getBoolean())
```

Comment:
The property is called `SAI_INDEX_READS_DISABLED` but this would act on _any_ kind of index: should we rather move this to `StorageAttachedIndex#isQueryable()` (which would also keep these methods consistent)?

Reply:
Fair point, I've moved the check.